app/services/katello/pulp3/repository.rb

```diff
@@ -386,6 +386,31 @@ def lookup_distributions(args)
         distributions_api.list(args).results
       end
 
+      def self.orphan_distributions(smart_proxy, pulp3_repo, options)
+        orphan_distributions_hrefs = []
+
+        distributions = fetch_from_list do |page_opts|
+          pulp3_repo.distributions_api.list(page_opts.merge(options))
+        end
+
+        distribution_hrefs = distributions.pluck(:_href)
+
+        orphan_distribution_hrefs = distribution_hrefs.select do |distribution_href|
+          dist = pulp3_repo.distributions_api.read(distribution_href)
+          is_orphan_distribution?(dist)
+        end
+      end
+
+      def self.is_orphan_distribution?(distribution)
+        if distribution.respond_to? :publication
+          distribution.publication.nil?
+        elsif distribution.respond_to? :repository and distribution.respond_to? :repository_version
+          distribution.repository.nil? && distribution.repository_version.nil?
+        else
+          false
+        end
```

Comment:
would something like:
```
distribution&.publication&.nil? && distribution&.repository&.nil? && distribution&.repository_version&.nil?
```
be better?

Reply:
Making that change doesn't pass the test written for it, unfortunately. But I'm still open to a more compact expression, of course.

Reply:
strange, i didn't realize &. didn't handle if the method wasn't there, what about:
```
        distribution.try(:publication).nil? && distribution.try(:repository).nil? && distribution.try(:repository_version).nil?
```
that seems to pass tests!